base_layer/core/src/chain_storage/lmdb_db/lmdb_db.rs

```diff
@@ -2013,6 +2013,31 @@ impl BlockchainBackend for LMDBDatabase {
         Ok(result)
     }
 
+    fn fetch_all_constitutions(
+        &self,
+        dan_node_public_key: &PublicKey,
+    ) -> Result<Vec<TransactionOutput>, ChainStorageError> {
+        let txn = self.read_transaction()?;
+        lmdb_filter_map_values(&txn, &self.utxos_db, |output: TransactionOutputRowData| {
```

Comment:
Fine for now, but we'll need to index this in the future

Reply:
Totally. I've made a note about the constitutions index for later but was less worried about it for the moment.